arangod/Replication2/Methods.cpp

```diff
@@ -202,23 +213,56 @@ struct ReplicatedLogMethodsCoordinator final
     return vocbase.getReplicatedLogs();
   }
 
-  auto getLogStatus(LogId id) const
+  [[noreturn]] auto getLocalStatus(LogId id) const
       -> futures::Future<replication2::replicated_log::LogStatus> override {
-    auto path = basics::StringUtils::joinT("/", "_api/log", id);
+    THROW_ARANGO_EXCEPTION(TRI_ERROR_NOT_IMPLEMENTED);
+  }
+
+  auto getGlobalStatus(LogId id) const
+      -> futures::Future<replication2::replicated_log::GlobalStatus> override {
+    auto path = basics::StringUtils::joinT("/", "_api/log", id, "local-status");
     network::RequestOptions opts;
     opts.database = vocbase.name();
-    return network::sendRequest(pool, "server:" + getLogLeader(id),
+    ServerID leaderId;
+
+    try {
+      leaderId = getLogLeader(id);
+    } catch (basics::Exception const& ex) {
+      if (ex.code() == TRI_ERROR_REPLICATION_REPLICATED_LOG_LEADER_RESIGNED) {
+        return GlobalStatus{
+            replication2::agency::methods::getCurrentSupervision(vocbase, id),
+            {},
+            std::nullopt};
+      }
+      throw;
+    }
```

Comment:
instead of dealing with this try..catch here you could use `clusterInfo.getReplicatedLogLeader(vocbase.name(), id)` directly, which returns a `ResultT<ParticipantId>`.